### ISSUE TITLE
AWSFacebookSignIn upgrade to depend on FBSDKCoreKit 5.x.x

### ISSUE THREAD

**Issue by BillBunting** [aws-amplify/aws-sdk-ios#1516]
**Which AWS Services is the feature request for?**
AWSFacebookSignIn (app using drop in UI)

**Is your feature request related to a problem? Please describe.**

There are new features in FBSDKShareKit that I would like to use, but since AWSFacebookSign in has a dependency on an older version, I am unable to use the new FBSDKShareKit features.

```
Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "FBSDKCoreKit":
  In Podfile:
    AWSFacebookSignIn (~> 2.9.7) was resolved to 2.9.7, which depends on
      FBSDKCoreKit (<= 4.42)

    FBSDKCoreKit (~> 4.42.0)

    FBSDKShareKit (~> 5.0.0) was resolved to 5.0.0, which depends on
      FBSDKCoreKit (= 5.0.0)
```

**Describe the solution you'd like**
Upgrade to Facebook FBSDKCoreKit 5.0.0 or newer.

**Describe alternatives you've considered**
Non known.  

Thank you.

**Comment by desokroshan** [comment]
This is being tracked in https://github.com/aws-amplify/aws-sdk-ios/issues/1397

**Comment by lis186** [comment]
Facebook login is broken on iOS 13. According  to the [response](https://github.com/facebook/facebook-swift-sdk/issues/482), update to the latest version of FBSDKCoreKit and FBSDKLoginKit (5.3.0) can fix the problem.

**Comment by BillBunting** [comment]
Current versions continue to move forward, but AWS `AWSFacebookSignIn` does not.  I'd like to be able to use current features in FBSDKShareKit but am unable because of `AWSFacebookSignIn`

```
- FBSDKCoreKit 4.42.0 -> 4.42.0 (latest version 5.7.0)
- FBSDKLoginKit 4.42.0 -> 4.42.0 (latest version 5.7.0)
- FBSDKShareKit 4.42.0 -> 4.42.0 (latest version 5.7.0)
```

```
[!] CocoaPods could not find compatible versions for pod "FBSDKCoreKit":
  In snapshot (Podfile.lock):
    FBSDKCoreKit (<= 4.42, = 4.42.0)

  In Podfile:
    AWSFacebookSignIn (~> 2.11.1) was resolved to 2.11.1, which depends on
      FBSDKCoreKit (<= 4.42)

    FBSDKShareKit (~> 5.6.0) was resolved to 5.6.0, which depends on
      FBSDKShareKit/Share (= 5.6.0) was resolved to 5.6.0, which depends on
        FBSDKCoreKit (~> 5.0)
```

**Comment by cornr** [comment]
We also would like to use Facebook SDK 5.X.X.

The failing stacktrace in our case is:

> 2019-10-16 12:11:27.288979+0200 App[24944:305553] -[FBSDKLoginManager logInWithReadPermissions:fromViewController:handler:]: unrecognized selector sent to instance 0x6000023ff480
2019-10-16 12:11:27.302801+0200 App[24944:305553] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[FBSDKLoginManager logInWithReadPermissions:fromViewController:handler:]: unrecognized selector sent to instance 0x6000023ff480'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23baa1ee __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff50864b20 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff23bcb154 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x00007fff23baef6c ___forwarding___ + 1436
	4   CoreFoundation                      0x00007fff23bb10f8 _CF_forwarding_prep_0 + 120
	5   AWSFacebookSignIn                   0x0000000111fc0dfb -[AWSFacebookSignInProvider login:] + 313
	6   AWSAuthCore                         0x0000000111c7c471 -[AWSSignInManager loginWithSignInProviderKey:completionHandler:] + 214
)

The 5.0.0 Release is almost half a year old. Would be really great if AWS could support it.

**Comment by BillBunting** [comment]
ecently, I upgraded to iOS 13 and Xcode 11 and found that AWSMobileClient's showSignIn() UI will not work until Facebook pods are upgrade to a min of FBSDKCoreKit 5.4.0 (current is 5.8) 

Facebook Sign-in UI Fails on iOS 13 / Xcode 11 (AWSMobileClient.default().showSignIn) Facebook SDK 5.8 required.  https://github.com/aws-amplify/aws-sdk-ios/issues/1974

I did manually modify the old version of the facebook code to make it work with AWSMobileClient as a workaround until the Amazon team is able to upgrade to the latest FBSDKs. iOS 13 requires ASWebAuthenticationPresentationContextProviding.

**Comment by cornr** [comment]
Thank you @kneekey23 for fixing this. Just wondering if you could to the same thing for the Google-SignIn-SDK 😎 https://github.com/aws-amplify/aws-sdk-ios/issues/1993

**Comment by kneekey23** [comment]
It’s on our backlog and we’ll update when we have more information. @cornr

**Comment by cornr** [comment]
Just tried 2.12.1 and FB SignIn is working again. Thanks.

**Comment by royjit** [comment]
Thank you @cornr for verifying. 

This has been released in the version [2.12.1](https://github.com/aws-amplify/aws-sdk-ios/releases/tag/2.12.1)
